### PR TITLE
fixes issue #43

### PR DESCRIPTION
**Pull request by shazqadeer**
This PR "fixes" issue #43 using the simple approach of returning `true` as the verification condition when the assertion count is zero.  This fix will still pay the cost of VC generation when there are no assertions but will avoid the cost of converting the VC to a string and sending it to the SMT solver.

### CODE REVIEW

**Comment by shazqadeer** [comment]
@zvonimir :  The test roundingmodes/FloatOpsWithoutRoundingMode.bpl  is failing in my PR.  I get the following output when I run the test in master.

```
Prover error: line 23 column 82: invalid number of arguments to floating point operator

Prover error: Prover died

FloatOpsWithoutRoundingMode.bpl(6,11): Verification inconclusive (foo)

Boogie program verifier finished with 0 verified, 0 errors, 1 inconclusive
```

But when I add an `assert true` at the end of `procedure foo`,  I get the following output which is different:
```
Prover error: line 22 column 94: sort mismatch, expected equal FloatingPoint sorts as arguments

Prover error: Prover died

FloatOpsWithoutRoundingMode.bpl(6,11): Verification inconclusive (foo)

Boogie program verifier finished with 0 verified, 0 errors, 1 inconclusive
```

What is the intention behind this test?  And is it ok to put `assert true` in `foo` and  change the expected output in the test so that it continues to pass?

File: Source/VCGeneration/VCGen.cs [hunk 1]
```diff
@@ -637,6 +637,10 @@ public VCExpr CodeExprToVerificationCondition(CodeExpr codeExpr, List<VCExprLetB
       }
 
       CumulativeAssertionCount += assertionCount;
+      if (assertionCount == 0)
+      {
+        return VCExpressionGenerator.True;
```
Comment (RustanLeino) [inline Source/VCGeneration/VCGen.cs:642]: So, after `True` is returned, what code prevents this trivial VC from being sent to the SMT solver?

Comment (shazqadeer) [reply Source/VCGeneration/VCGen.cs:642]: The VC will be sent to the SMT solver.   My PR description may be misleading.  I only wanted to claim that the original VC is not sent to the SMT solver.  But a trivial VC will be sent.

**Comment by shazqadeer** [comment]
@zvonimir :  I added an `assert true` at the beginning of foo.  This helps maintain the test.

**Comment by zvonimir** [comment]
@shazqadeer I looked at what is happening in ` roundingmodes/FloatOpsWithoutRoundingMode.bpl`. This regression is testing that the prover fails when floating-points operations with wrong numbers of arguments are passed to it. which can happen if a user is not careful when using `builtin` attribute. But I think this is something we do not need a regression for, and so I suggest you remove this regression.

File: Source/VCGeneration/VCGen.cs (hunk 1 above)
Comment (shazqadeer) [reply Source/VCGeneration/VCGen.cs:642]: @RustanLeino : Does this fix look reasonable to you?

**Review by RustanLeino — APPROVED** [review]

**Pull request merged**